### PR TITLE
[now-build-utils] Prefer yarn over npm install when both detected

### PR DESCRIPTION
**Pull request by styfle**
Previously, when `package-lock.json` was detected, we would always use `npm install`.

This PR changes the behavior to the following:
- If only `package-lock.json` is detected, we use `npm install`.
- If both `package-lock.json` and `yarn.lock` are detected, use `yarn install`.

This will avoid npm bugs such as `npm ERR! Cannot read property 'match' of undefined`.

### CODE REVIEW

File: packages/now-build-utils/src/fs/run-user-scripts.ts [hunk 1]
```diff
@@ -200,9 +201,16 @@ async function scanParentDirs(destPath: string, readPackageJson = false) {
         packageJson = JSON.parse(await fs.readFile(packageJsonPath, 'utf8'));
       }
       // eslint-disable-next-line no-await-in-loop
-      hasPackageLockJson = await fs.pathExists(
+      const hasPackageLockJson = await fs.pathExists(
         path.join(currentDestPath, 'package-lock.json')
       );
+      // eslint-disable-next-line no-await-in-loop
+      const hasYarnLock = await fs.pathExists(
```
Comment (TooTallNate) [inline packages/now-build-utils/src/fs/run-user-scripts.ts]: Let's check these files in parallel using `Promise.all()`?

File: packages/now-build-utils/src/fs/run-user-scripts.ts [hunk 2]
```diff
@@ -211,7 +216,7 @@ async function scanParentDirs(destPath: string, readPackageJson = false) {
     currentDestPath = newDestPath;
   }
 
-  return { hasPackageLockJson, packageJson };
+  return { cliType, packageJson };
```
Comment (TooTallNate) [inline packages/now-build-utils/src/fs/run-user-scripts.ts:219]: I think we should keep `hasPackageLockJson` around for backwards-compat purposes.

Comment (styfle) [reply packages/now-build-utils/src/fs/run-user-scripts.ts:219]: The function `scanParentDirs()` is not exported. All the usages are in this file.

What backwards-compat use case are you thinking of?

Comment (TooTallNate) [reply packages/now-build-utils/src/fs/run-user-scripts.ts:219]: Oh ok, in that case 👍

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**